### ISSUE TITLE
Set loopback port(?) on linux for debugging.

### ISSUE THREAD

**Issue by bsutton** [Bdaya-Dev/oidc#95]
I'm not certain if this is a bug or I just don't understand how to configure the auth.

My problem is that in my dev environment (linux) oidc is trying to listen on 127.0.0.1:443.

443 is a privileged port so this fails. As this is a debug environment I can't run under sudo without screwing a whole lot of things up.

I've found the example code  but I don't quite understand what it is doing with the redirect url.

For linux the example suggests: 
```dart
OidcUserManagerSettings(
          scope: _scopes,
          redirectUri:  Platform.isWindows || Platform.isLinux
                      ? Uri.parse('http://localhost:0')
                      : Uri(),
        ));

```
However this fails;  when the request is sent to the identify provider (xero.com) it reports an error that the redirect url is invalid.

When I change the redirect url to:

```dart
OidcUserManagerSettings(
          scope: _scopes,
          
          redirectUri:  Platform.isWindows || Platform.isLinux
               Uri.parse('http://ivanhoehandyman.com.au/app_auth_redirect.html')),
                      : Uri(),
        ));
````

The identity provider (xero.com) is now happy with the url, but the dart code then tries to open a socket on 127.0.0.1:443.

So how do I make the identity provider happy and get the oidc code to not listen on port 443 in debug mode (or at all).
I'm also not certain why its listening on 127 - is it try to provide a local server for the browser to redirect to?
As my redirect is hosted this should be required.

Help.

**Comment by ahmednfwela** [comment]
Hi @bsutton, Thanks for bringing this to my attention.
The code responsible for handling desktop auth is mostly located here:

[package:oidc_desktop](https://github.com/Bdaya-Dev/oidc/blob/main/packages/oidc_desktop/lib/src/oidc_desktop.dart) and [package:oidc_loopback_listener](https://github.com/Bdaya-Dev/oidc/blob/main/packages/oidc_loopback_listener/lib/src/oidc_loopback_listener.dart)

In summary, it implements approach [rfc8252 Section 7.3](https://datatracker.ietf.org/doc/html/rfc8252#section-7.3) where the app uses a a "randomly assigned port" to listen to redirect response from the browser, which [-as stated in the RFC-](https://datatracker.ietf.org/doc/html/rfc8252#section-8.1) is the most secure.

so back to your issue, we use [HttpServer.bind](https://api.dart.dev/stable/3.4.4/dart-io/HttpServer/bind.html) to reserve the port and start the server, which as stated in the docs:

> If port has the value 0 an ephemeral port will be chosen by the system.

so you can select the port yourself by changing the redirect uri to a known empty port:

```dart
OidcUserManagerSettings(
  scope: _scopes,
  redirectUri:  Platform.isWindows || Platform.isLinux
              ? Uri.parse('http://localhost:12335')
              : Uri(),
)
```

On my side though, what I can do is improve port selection process to skip any privileged ports

**Comment by bsutton** [comment]
OK, apart from the problem you noted with xero not taking a random port, my issue was that I had registered an external redirect url:
http://ivanhoehandyman.com.au/app_auth_redirect.html

Once I added http://localhost:12335 to the list of valid redirect urls in xero the problem was resolved.

My reading had lead me to believe that xero required the redirect to be hosted on a public web site.  But I had clearly misunderstood.

Appreciate the help!!
